submitter/relayer/relayer.go

```diff
@@ -81,25 +95,68 @@ func (rl *Relayer) SendCheckpointToBTC(ckpt *ckpttypes.RawCheckpointWithMetaResp
 		return nil
 	}
 
+	storeCkptFunc := func(tx1, tx2 *wire.MsgTx, epochNum uint64) error {
+		storedCkpt := store.NewStoredCheckpoint(tx1, tx2, epochNum)
+		return rl.store.PutCheckpoint(storedCkpt)
+	}
+
 	if rl.shouldSendCompleteCkpt(ckptEpoch) {
+		hasBeenProcessed, err := maybeResendFromStore(
+			ckptEpoch,
+			rl.store.LatestCheckpoint,
+			rl.GetRawTransaction,
+			rl.sendTxToBTC,
+		)
+		if err != nil {
+			return err
+		}
+
+		if hasBeenProcessed {
+			return nil
+		}
```

Comment:
seems this chunk of code is irrelevant to whether we should send a complete ckpt or only the second half. Can we move it out to avoid duplicate code?